---
title: "AlgEpi question catalog"
author: "Johannes Hausmann, Luis Kress"
date: "`r Sys.Date()`"
#knit: pagedown::chrome_print
links-to-footnotes: true
output:
  html_document:
    css: css/test.css
---

> Written and answered on a best effort basis. It might contain wrong answers.
> Copying without questioning leads to your ruin!

## 1. Introduction into Epigenetics

### What is epigenetics?

* studies on heritable changes
  + of gene expression
  + of cellular phenotype
  
* caused by mechanisms other than changes in the nucleotide sequence

### What happens in DNA methylation?

* methylation of the 5th C-atom of cytosine
  + 5-Methylcytosine

* occurring in the CpG context
  +   both strands of the DNA are recognized and methylated

* dynamic process
  + active methylation (_de novo_: DNMT3a/b, conservation: DNMT1)
  + de-methylation (TET1/2/3, BER)


![Methylation – De-methylation path of cytosine](media/lecture1_methylation.jpg){width=50%}

* DNA methylation is also used to define the parental strand during replication

### How can DNA methylation affect gene expression?


* CpG islands are cytosine and guanin rich regions in the genome

* conserved region in or close to the promoter

* unmethylated = gene is active
  + TF can bind &rarr; gene is expressed

* methylated = gene is silenced
  + TF can not bind &rarr; gene not expressed
  
* Methylated DNA can interfere (block or enhance) the binding of
transcription factors

* methyl binding proteins
  + recognize methylated CpG sites
  + recruit histone modifying enzymes &rarr; chromatin changes


### What are histone modifcations?

* DNA is wrapped around Histones (protein of a certain type)
  + DNA + core histones &rarr; nucleosome

* Addition of H1 (linker histone) &rarr; 30nm fibre (less active genes)

* Biochemical modifications of core histones
  + influence affinity of linker histone (H1)  
  &rarr; consequence is change in chromatin structure  
  

* Histone tails (N-terminal) get modified
  + position and type influence the outcome (activation or repression of transcription)
  
![Type of histone modifications and naming convention](media/lecture1_histone_modification.jpg){width=50%}


### Which histone modifications are associated with active transcription or heterochromatin?

* Acetylation of histones at N-terminal ends is associated with transcriptional activity

* Lysins in terminal ends lead to positive charge that interacts with negative charge of DNA
  
* Acetylated lysins reduces positive charge
  + Loss of affinity to DNA

* Change of charge also effects interaction of nucleosomes among each other
    + Loss of 30nm filaments
    
* Good overview: [Link to overview table (https://doi.org/10.1186/s13059-019-1870-5)](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1870-5/tables/1)


### What is Waddington‘s epigenetic landscape model?

* 1957: Conrad Waddington's concept of an epigenetic landscape to represent the process of cellular decision-making during development. At various points in this dynamic visual metaphor, the cell (represented by a ball) can take specific permitted trajectories, leading to different outcomes or cell fates.

* Dellular differentiation can be considered an epigenetic phenomenon, largely governed by changes in “epigenetic landscape” rather than alterations in genetic inheritance

![Waddington's Classical Epigenetic Landscape](media/lecture1_waddington.jpg){width=50%}


* Chromatin structure of germ cells loosened and DNA
methylation is reduced to a minmum &rarr; totipotent cell types

* cell-type specific pattern established in following cell division cycles &rarr; cell-type specific gene expression

### What is imprinting?

* Allele-specific expression of a genev (either from the maternal or paternal allele) via CpG methylation

### Explain how mosaic colouring of cat fur comes about?

* Random activation or inactivation during embryonic development of one of the X chromosomes, where the responsible gene for fur coloring is located

## 2. Understanding gene expression with RNA sequencing

### What is RNA-seq?

* Sequencing of mature mRNA molecules

* Fragmentation of transcripts and reverse transcription to cDNA (hexamer priming)

* Generation of ds-cDNA fragments and high throughput sequencing

* Can be used to estimate gene expression or identify novel transcriptional changes


![RNA-seq experiment workflow](media/lecture2_rna_seq.png){width=50%}

### How to obtain RNA counts?

* Two strategies to map and quantify RNA-seq reads

1. Mapping to the transcriptome  
    + Reduced search space
    + Can only assign reads if they originate from known transcripts
    + No novel transcripts

2. Mapping to the genome  
    + Large search space for alignment
    + Methods need to be splice-aware
    + _in silico_ splice reads to consider introns
  
* Transcript counts are aggregated to obtain gene level counts

* Genome mapping &rarr; count reads per gene bin 

![RNA-seq counts](media/lecture2_rna_count.png){width=50%}

### Why do we need to normalize RNA-seq read counts?
 
* gene expression measurement
  + to compare expression between genes
  + to compare expression of a gene between samples

* An estimation of the average copy number of a gene‘s RNAs

* We need to consider
  + Differences in gene length  
  &rarrw; longer genes produce more fragments  
  
  + Differences in library size
    + Number of reads obtained per experiment by the sequencer is different

### What are TPM and RPKM normalization?

* Metrics to adjust raw read counts for gene length and sequencing depth

* Very similar but calculated in different orders

* RPKM (Reads Per Kilobase per Million Mapped reads)
  + single end sequencing &rarr; FPKM for paired end reads
  + Scale read counts by number of reads mapped and multiplied with scaling factor (10^9)

### Why is TPM better?

### How to quantify transcript expression?

### What is the effective length of a transcript?

### Which biases does Salmon take into account?

## 3. Analysis of differential expression data and interpretation

### Which normalization does DEseq2 use?

### What is the advantage of the negative binomial compared to the Possion distribution, when looking at counts?

### What is a p-value?

### What does FDR mean

### Why do I need to do FDR?

### How does DEseq2 fit the dispersion for genes?

### What is an MA-plot, and how to read it?

### What is the Gene Ontology?

### How is functional enrichment estimated?

## 4. Analysis of ChIP-seq data

### What is ChIP-seq?

### Why do we need to normalize ChIP-seq read counts?

### What is indirect binding?

### Explain basic quality steps for ChIP-seq data?

### How does peak calling with MACS2 work?

### What are the 6 core histone modifications?

### How can ChIP-seq data be used to locate tissue-specifc enhancers?

### Mention approaches for differential peak calling!

## 5. Analysis of DNA methylation data

### Why does sodium bisulfite treatment allow DNA methylation analysis?

### Why is WGBS called the gold standard in DNA methylation analysis?

### Why is bisulfite-seq data difficult to analyze?

### What is the asymmetric mapping problem?

### How do reduced representation mappers perform alignment?

### What are PMDs and how does MethylSeekR find them?

### How does RnBeads do differential methylation detection?

### Why is epigenetics on cell-free DNA interesting?

## 6. Analysis of open-chromatin data

### What does open-chromatin mean?

### Which assays are commonly used to profile open-chromatin regions?

### How is NOMe-seq different to other assays?

### What is the underlying idea of the gNOMeHMM approach for?

### How does JAMM use mixture models for detecting peaks?

### Which aspect of ATAC-seq data is specifically modelled by the HMM in the HMMRATAC method?

### Which of the two assays are more similar ATAC-, NOMe- and, DNase1-seq and why?

### How can you compute overlap between peak datasets?

* Since peaks are just intervalls of genomic regions you can use BEDtools
to intersect the region to find overlaps





